pkg/model/multiversion/manager.go

```diff
@@ -0,0 +1,197 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package multiversion
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"gopkg.in/src-d/go-git.v4"
+
+	ackgenconfig "github.com/aws-controllers-k8s/code-generator/pkg/generate/config"
+	ackmodel "github.com/aws-controllers-k8s/code-generator/pkg/model"
+	"github.com/aws-controllers-k8s/code-generator/pkg/util"
+)
+
+var (
+	ErrAPIVersionNotFound   = errors.New("api version not found")
+	ErrAPIVersionRemoved    = errors.New("api version removed")
+	ErrAPIVersionDeprecated = errors.New("api version deprecated")
+)
+
+// APIVersionManager is a API versions manager. It contains the mapping
+// of each non-deprecated version with their correspending ackmodel.Model
+// and APIInfos.
+type APIVersionManager struct {
+	gitRepo *git.Repository
+
+	hubVersion    string
+	spokeVersions []string
+
+	deprecatedVersions []string
+	removedVersions    []string
+
+	apiInfos map[string]APIInfo
+	models   map[string]*ackmodel.Model
+}
+
+// NewAPIVersionManager initialises and returns a new APIVersionManager.
+func NewAPIVersionManager(
+	sdkCacheDir string,
+	serviceAlias string,
+	hubVersion string,
+	apisInfo map[string]APIInfo,
+	defaultConfig ackgenconfig.Config,
+) (*APIVersionManager, error) {
+	if len(apisInfo) == 0 {
+		return nil, fmt.Errorf("empty apisInfo")
+	}
+
+	spokeVersions := make([]string, 0, len(apisInfo)-1)
+	gitRepo, err := util.LoadRepository(sdkCacheDir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read sdk git repository: %v", err)
+	}
```

Comment:
Do we have tests that we can identify and load the repository correctly? All the delta tests use the same AWS SDK version

Reply:
> Do we have tests that we can identify and load the repository correctly?

Unfortunately no.. the testdata we have is a collection of text files and they are not part of an `aws-sdk-go` git repository.

> All the delta tests use the same AWS SDK version

I just added new delta unit tests using multiple AWS SDK versions